dot.files/bin/vol

```diff
@@ -1,15 +1,15 @@
-#!/bin/sh
+#!/bin/bash
 
 export XDIALOG_NO_GMSGS=1
 
 # Pick a default available UI ...
-if [[ -x "$(which Xdialog 2> /dev/null)" && -n "${DISPLAY}" ]]; then # Check for X, Xdialog
-    declare DIALOG=$(which Xdialog) G=("100x75")
-elif [[ -x "$(which dialog 2> /dev/null)" ]]; then # Check for dialog
-    declare DIALOG=$(which dialog) G=( 0 0 )
+if type -fP Xdialog > /dev/null 2>&1 && [ -n "${DISPLAY}" ]; then # Check for X, Xdialog
+    declare DIALOG="$(type -fPp Xdialog)" G=("100x75")
+elif type -fP dialog > /dev/null 2>&1; then # Check for dialog
+    declare DIALOG="$(type -fPp dialog)" G=( 0 0 )
```

Comment:
I'd argue `type` is a better choice over `which` (I used to use `which`, too) as it's a builtin and requires no additional process. See `help type` for information on the flags I used.